
# Error while splitting and exploding into different rows . Non unique multi-index error

I have to explode multiple columns into separate rows which are in a list. For example in below dataset:
  import pandas as pd
  df = pd.DataFrame({'var1': ['[2.0, 2.0, 2.0, 2.0]', '[1.0, 5.0]'], 
                     'var2': ['[A4003250601, A4003250601, A4003250601, 003250601]','[A4002000401, A9262000201]'], 
                     'var3': [1, 2]})

  dataframe
     var1                                               var2                var3
0  [2.0, 2.0, 2.0, 2.0]  [A4003250601, A4003250601, A4003250601, A40032...     1
1            [1.0, 5.0]                         [A4002000401, A9262000201]     2

Is correctly getting exploded when using the below code :
(df.set_index(['var3']) 
       .apply(lambda col: col.astype(str).str.split(',')
       .explode())
       .reset_index()
       .reindex(df.columns, axis=1))

Output: (when index column in set to var3)

    var1           var2  var3
0   [2.0   [A4003250601     1
1    2.0    A4003250601     1
2    2.0    A4003250601     1
3   2.0]   A4003250601]     1
4   [1.0   [A4002000401     2
5   5.0]   A9262000201]     2

But when I use it in a datafame where there are multiple index columns and some of them have duplicates it gives me a ValueError, Error: cannot handle a non-unique multi-index!. I have to use those columns so not sure how to take the index without the error.
Dataframe
df = pd.DataFrame({'var1': ['[2.0, 2.0, 2.0, 2.0]', '[1.0, 5.0]', '[2.0, 2.0, 2.0, 2.0]', '[6.0]', '[2.0, 5.0]', '[2.0, 2.0, 2.0, 2.0]', '[3.0]', '[1.0, 5.0]'], 
                   'var2': ['[A4003250601, A4003250601, A4003250601, A4003250601]','[A4002000401, A9262000201]', '[A4003250601, A4003250601, A4003250601, A4003250601]', '[A4001800801]', '[A4002000401, A9262000201]', '[A4003250601, A4003250601, A4003250601, A4003250601]', '[A4001800801]', '[A4002000401, A9262000201]'], 
                   'var3': ['[335.764, 335.764, 335.764, 335.764]', '[191.0, 310.0]', '[335.764, 335.764, 335.764, 335.764]', '[240.0]', '[191.0, 310.0]', '[335.764, 335.764, 335.764, 335.764]', '[240.0]', '[191.0, 310.0]'], 
                   'var4':['[15659839, 15659839, 15659839, 15659839]', '[15659891, 15659891]', '[15659839, 15659839, 15659839, 15659839]', '[15659891]', '[15659891, 15659891]', '[15659839, 15659839, 15659839, 15659839]', '[15659891]', '[15659891, 15659891]'], 
                   'var5':['[PMAC, PMAC, PMAC, PMAC]', '[SUNDARAM FASTENERS LTD (GUMMIDIPOONDI), AUTOLEC DIVISION , SUNDARAM FASTENERS LTD (GUMMIDIPOONDI), AUTOLEC DIVISION ]', '[PMAC, PMAC, PMAC, PMAC]', '[SUNDARAM FASTENERS LTD (GUMMIDIPOONDI), AUTOLEC DIVISION ]', '[SUNDARAM FASTENERS LTD (GUMMIDIPOONDI), AUTOLEC DIVISION , SUNDARAM FASTENERS LTD (GUMMIDIPOONDI), AUTOLEC DIVISION ]', '[PMAC, PMAC, PMAC, PMAC]', '[SUNDARAM FASTENERS LTD (GUMMIDIPOONDI), AUTOLEC DIVISION ]', '[SUNDARAM FASTENERS LTD (GUMMIDIPOONDI), AUTOLEC DIVISION , SUNDARAM FASTENERS LTD (GUMMIDIPOONDI), AUTOLEC DIVISION ]'],
                   'var6': [20221031,20221031,20221101, 20221101, 20221101, 20221102,20221102,20221102],
                   'var7':[0.8544,0.42, 0.8544, 0.729,0.42, 0.8544, 0.729, 0.42],
                   'var8':[1200,1000,1200,900,1000,1200,900,1000]})


Code to split into new rows
(df.set_index(['var6', 'var7', 'var8']) 
       .apply(lambda col: col.astype(str).str.split(',')
       .explode())
       .reset_index()
       .reindex(df.columns, axis=1))

This gives a ValueError: cannot handle a non-unique multi-index!

        